hybrid/decomposers.py

```diff
@@ -725,31 +729,34 @@ def next(self, state, **runopts):
         return state.updated(subproblem=subbqm)
 
 
-def _all_minimal_covers(edgelist):
-    """Create all minimal covers, provided graphs contain fewer than 17 edges."""
-
-    if len(edgelist) > 16:
-        raise ValueError(
-            f'len(edgelist) = {len(edgelist)} > 16. Exponential scaling of this method means it '
-            'should only be used for small graphs. The use case for this function '
-            'is in finding coverings over unyielded lattice subgraphs, which in '
-            'the current online generation of processors are sufficiently small '
-            'so as to allow a brute force method.')
-
-    minimum_coverings = []
-    for cover in map(set, product(*edgelist)):
-        newmin = []
-        for s in minimum_coverings:
-            if s < cover:
-                break
-            elif cover < s:
-                pass
-            else:
-                newmin.append(s)
+def _good_cover(edgelist, brute_force_threshold=16):
+    # Attempt to solve by brute force (tree decomposition, under tree decomposition)
+    try:
+        tds = TreeDecompositionSolver()
+        G = nx.from_edgelist(edgelist)
+        coverLTW = dnx.algorithms.cover.min_vertex_cover(G=G, sampler=tds)
+        return coverLTW
+    except:
+        warnings.warn('A verifiable minimum cover is not found '
+                      'a simple efficient heuristic is used; '
+                      'Additional care may be warranted.')
+
+        # Choose best of two simple deterministic heuristics, randomized
+        # heuristics (like QA) will typically work better in practice:
+        G = nx.from_edgelist(edgelist)
+        cover1 = min_weighted_vertex_cover(G) # A networkx default
+        G = nx.from_edgelist(edgelist)
```

Comment:
Are you creating a new `G` because the `G` from above is mutated by `min_weighted_vertex_cover(G)`? The [docs](https://networkx.org/documentation/stable/reference/algorithms/generated/networkx.algorithms.approximation.vertex_cover.min_weighted_vertex_cover.html) don't mention that.